### PR TITLE
core: Remove support for ceph pacific

### PR DESCRIPTION
**Pull request by travisn**
Pacific is end of life and no longer necessary to support in Rook with v1.13.
Remove all the examples of Pacific and the special code considerations for Pacific.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/daily-nightly-jobs.yml [hunk 1]
```diff
@@ -395,7 +359,7 @@ jobs:
     strategy:
       matrix:
         ceph-image-tag:
-          ["latest-main-devel", "latest-quincy-devel", "latest-pacific-devel"]
+          ["latest-main-devel", "latest-quincy-devel", "latest-reef-devel"]
```
Comment (BlaineEXE) [inline .github/workflows/daily-nightly-jobs.yml:362]: The CI title says `upgrade-from-quincy-stable-to-quincy-devel`. I guess this wasn't right before also?

```suggestion
          ["latest-main-devel", "latest-quincy-devel", "latest-quincy-devel"]
```

File: Documentation/Upgrade/ceph-upgrade.md [hunk 2]
```diff
@@ -24,14 +24,13 @@ until all the daemons have been updated.
...
+Rook v1.13 supports the following Ceph versions:
 
 * Ceph Reef v18.2.0 or newer
 * Ceph Quincy v17.2.0 or newer
-* Ceph Pacific v16.2.7 or newer
 
-Support for Ceph Pacific (16.2.x) will be removed in the next Rook release. It will be mandatory to
-upgrade to Quincy or Reef before upgrading to the Rook release after v1.12.x.
+Support for Ceph Pacific (16.2.x) is removed in Rook v1.13. Upgrade to Quincy or Reef before upgrading
+to Rook v1.13.
```
Comment (BlaineEXE) [inline Documentation/Upgrade/ceph-upgrade.md:33]: I wonder if this should be a `!!! warn` to make it more attention grabbing?

File: pkg/operator/ceph/object/topic/provisioner.go [hunk 3]
```diff
@@ -166,12 +165,8 @@ func createTopicAttributes(topic *cephv1.CephBucketTopic, cephVersion version.Ce
...
-			cloudEvents := strconv.FormatBool(topic.Spec.Endpoint.HTTP.SendCloudEvents)
-			attr["cloudevents"] = &cloudEvents
-		} else if topic.Spec.Endpoint.HTTP.SendCloudEvents {
-			logger.Warning("HTTP CloudEvents endpoint supported for Ceph Quincy (v17) or newer")
-		}
+		cloudEvents := strconv.FormatBool(topic.Spec.Endpoint.HTTP.SendCloudEvents)
+		attr["cloudevents"] = &cloudEvents
 	}
 	if topic.Spec.Endpoint.Kafka != nil {
 		logger.Infof("creating CephBucketTopic %q with endpoint %q", nsName, topic.Spec.Endpoint.Kafka.URI)
```
Comment (BlaineEXE) [inline pkg/operator/ceph/object/topic/provisioner.go:172]: Might be good to have @yuvalif review this just in case

**Review by BlaineEXE — COMMENTED** [review]
Looks good. Just a few questions.

File: .github/workflows/daily-nightly-jobs.yml (hunk 1 above)
Comment (travisn) [reply .github/workflows/daily-nightly-jobs.yml:362]: This is under the `rgw-multisite-test-with-ceph-devel`, right? So the test was missing reef tests previously.

File: Documentation/Upgrade/ceph-upgrade.md (hunk 2 above)
Comment (travisn) [reply Documentation/Upgrade/ceph-upgrade.md:33]: Agreed, we will need to make it obvious in the release notes, release blog, warnings in the breaking changes, etc. But I'd suggest we evaluate the warning when we do a more thorough review of the upgrade guide.

**Review by subhamkrai — APPROVED** [review]
LGTM

**Review by BlaineEXE — APPROVED** [review]

**Pull request merged**